xblockutils/templatetags/i18n.py

```diff
@@ -44,6 +45,8 @@ def merge_translation(self, context):
         # Revert to original translation object
         if language in self._translations:
             trans_real._translations[language] = self._translations[language]
+            # Re-activate the current language to resets translation caches
+            trans_real.activate(language)
```

Comment:
@mtyaka I found a bug with my latest approach -- the translation change wasn't getting reverted completely, and so the XBlock translations could bleed into other places in the platform.  

This line ensures that [`trans_real._active.value`](https://github.com/django/django/blob/1.8.19/django/utils/translation/trans_real.py#L213-L225) is reverted too.

Updated the tests to move the "untranslated text" check into the same method as the "translated text" check, which demonstrates that the translation was fully reverted.